src/accountsdb/db.zig

```diff
@@ -170,14 +170,18 @@ pub const AccountsDB = struct {
         number_of_index_shards: usize,
         /// Amount of BufferPool frames, used for cached reads. Default = 1GiB.
         buffer_pool_frames: u32 = 2 * 1024 * 1024,
+        /// For supplying your own ReferenceAllocator. Required for index_allocation = .parent.
+        injected_ref_allocator: ?*sig.accounts_db.index.ReferenceAllocator = null,
```

Comment:
Would suggest something akin to:
```zig
index_allocation: union(AccountIndex.AllocatorConfig.Tag) {
    ram,
    disk,
    parent: *sig.accounts_db.index.ReferenceAllocator,
},
```
to make the invalid parameter combination unrepresentable.

Reply:
Changed in ae65af4b8667ee59bfc4c81a81d8f5f835493888